### DISCUSSION TITLE
Table of figures/tables fail to generate

### DISCUSSION THREAD

**Discussion by warnes** [davidgohel/officedown#107] [General]
I have installed the the latest version of officedown from GitHub on MacOS 13 with Microsoft Word 16.71.

When I run the provided example:
```
dir <- system.file(package = "officedown", "examples", "bookdown")
file.copy(dir, getwd(), recursive = TRUE, overwrite = TRUE)
rmarkdown::render_site("bookdown")
browseURL("bookdown/_book/bookdown.docx")
```
the document is generated without error.  However, when I open the document, 
* I get a dialog box 'This document contains fields that may refer to other files.  Do you want to update the fields in the document'.
* I click 'Yes'.
* I then get the message "Creating a table of contents?  Start by applying a header style from the Styles gallery to the selected text" *twice*:
![image](https://user-images.githubusercontent.com/6144863/234354316-34475a97-cb79-435b-9845-c53ab48eb680.png)
* I click OK to both dialog boxes.

The resulting documents does have a proper table of contents, but no Table of figures or Table of tables:
![image](https://user-images.githubusercontent.com/6144863/234354814-0af9d71d-878f-482b-96f0-8c8f57e7a266.png)

**Comment by davidgohel** [comment]
I don't get the same behavior. On my machine, it works as expected. 

We have written 'doconv' for this type of issue: when the job must be done by Word. Here, it needs to update the computed fields, toc are Word computed fields and their content is populated by Word , not officer.

```
dir <- system.file(package = "officedown", "examples", "bookdown")
file.copy(dir, getwd(), recursive = TRUE, overwrite = TRUE)
rmarkdown::render_site("bookdown")
doconv::docx_update("bookdown/_book/bookdown.docx")
```

`doconv::docx_update()` will run "select all + CTRL F9", so update all computed fields in the document.

As you are on mac, be sure to init (only once) permissions to write - see `?doconv::working_directory`

**Comment by warnes** [comment]
Hi David, thanks for responding.

Unfortunately, following your instructions does not resolve the problem.

However, I by manually adding a table of figures and a table of tables to the document, I discovered that the field codes differ from what is present in the example bookdown document:

In the bookdown example document, the field codes are:
```
{TOC \h \z \t "Image Caption;1"}
{TOC \h \z \t "Table Caption; 1"}
```
the codes inserted by manually adding the TOF and TOT are:
```
{TOC \h \z \c "Figure"}
{TOC \h \z \c "Table"}
```

So, I tried modifying `02-toc.Rmd` to explicitly specify these codes:
![image](https://user-images.githubusercontent.com/6144863/236876844-8715f1a7-80ca-4d23-a16a-6c9f5aa96c2b.png)


Now, running 
 ```r
unlink("bookdown/_book/", recursive = TRUE)
rmarkdown::render_site("bookdown")
doconv::docx_update("bookdown/_book/bookdown.docx")
system("open bookdown/_book/bookdown.docx")
```

Yields the correct output:
![image](https://user-images.githubusercontent.com/6144863/236877263-d3f7818a-5882-4c99-b41f-adaefaa5fe59.png)

**Comment by warnes** [comment]
Looking into the code of `officer:: to_wml.block_toc`, it looks like the right thing to do is to specify `seq_id` instead of `style`:

![image](https://user-images.githubusercontent.com/6144863/236910588-6d3333b5-8bc1-4de5-a4d4-f9b90c83a006.png)

**Comment by cbusath** [comment]
I am having a similar issue, though for me it seems to be that flextables are not being recognized. I also tried gt tables. Is this expected? 


```
---
output: officedown::rdocx_document
---
```
```
```{r message = F, warning = F, echo = F}
library(tidyverse)
library(officedown)
library(flextable)

\```

# List of Figures

<!---BLOCK_TOC{seq_id: "fig"}--->

# List of Tables

<!---BLOCK_TOC{seq_id: "tab"}--->

```{r tab.cap = "Flextable"}
mtcars[1:2,] %>% 
  flextable()
\```

```{r tab.cap = "Normal Table"}
mtcars[1:2,]
\```

```{r fig.cap = "Plot 1"}

mtcars %>% 
  ggplot(aes(x = mpg, y = cyl)) + 
  geom_point()
\```

```

First, I get a similar request about updating from another file:
![image](https://github.com/davidgohel/officedown/assets/106763476/81d75457-8d59-429a-976b-366b31c5befb)



This gives me something that looks like this (1st page, 2nd page has a plot):

![image](https://github.com/davidgohel/officedown/assets/106763476/bc486a2d-163d-4861-b5f5-2608b22487b0)
